source/interface.cpp

```diff
@@ -146,6 +146,38 @@ void webEnginePageRunJavaScript(void* webEnginePage, char* javascript, int id, f
     _webEnginePage->runJavaScript(javascript, [id, callback](const QVariant &v) { callback(id, v.toString().toUtf8().data()); });
 }
 
+void webEnginePageSetLivecycleState(void* webEnginePage, int state) {
+    QWebEnginePage *_webEnginePage = reinterpret_cast<QWebEnginePage*>(webEnginePage);
+    _webEnginePage->setLifecycleState(static_cast<QWebEnginePage::LifecycleState>(state));
+}
+
+int webEnginePageLivecycleState(void* webEnginePage) {
+    QWebEnginePage *_webEnginePage = reinterpret_cast<QWebEnginePage*>(webEnginePage);
+    return (int)_webEnginePage->lifecycleState();
+}
+
+void webEnginePageTriggerAction(void* webEnginePage, int action, int checked) {
+    QWebEnginePage *_webEnginePage = reinterpret_cast<QWebEnginePage*>(webEnginePage);
+    _webEnginePage->triggerAction(static_cast<QWebEnginePage::WebAction>(action), checked);
+}
+
+void* webEnginePageAction(void* webEnginePage, int action) {
+    QWebEnginePage *_webEnginePage = reinterpret_cast<QWebEnginePage*>(webEnginePage);
+    return _webEnginePage->action(static_cast<QWebEnginePage::WebAction>(action));
```

Comment:
Make sure to cast things to `void*` if the return type is `void*`. Some compilers can warn or break because of that.

Reply:
Good point!